core/src/main/java/org/kiwix/kiwixmobile/core/page/bookmark/BookmarksActivity.kt

```diff
@@ -1,149 +1,57 @@
-  val compositeDisposable = CompositeDisposable()
-  private val bookmarkViewModel by lazy { viewModel<BookmarkViewModel>(viewModelFactory) }
+class BookmarksActivity : PageActivity() {
+  override val pageViewModel by lazy { viewModel<BookmarkViewModel>(viewModelFactory) }
 
-  private val actionModeCallback: ActionMode.Callback =
-    object : ActionMode.Callback {
-      override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
-        mode.menuInflater.inflate(R.menu.menu_context_delete, menu)
-        return true
-      }
-
-      override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean = false
-
-      override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
-        if (item.itemId == R.id.menu_context_delete) {
-          bookmarkViewModel.actions.offer(Action.UserClickedDeleteSelectedPages)
-          return true
-        }
-        bookmarkViewModel.actions.offer(Action.ExitActionModeMenu)
-        return false
-      }
-
-      override fun onDestroyActionMode(mode: ActionMode) {
-        bookmarkViewModel.actions.offer(Action.ExitActionModeMenu)
-        actionMode = null
-      }
-    }
-
-  override fun injection(coreComponent: CoreComponent) {
-    activityComponent.inject(this)
-  }
-
-  private val bookmarksAdapter: BookmarksAdapter by lazy {
-    BookmarksAdapter(BookmarkItemDelegate(this))
+  override val pageAdapter: PageAdapter by lazy {
+    PageAdapter(BookmarkItemDelegate(this))
   }
 
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_bookmarks)
-    setSupportActionBar(toolbar)
-
-    supportActionBar?.setDisplayHomeAsUpEnabled(true)
-    supportActionBar?.setTitle(R.string.bookmarks)
-
-    recycler_view.adapter = bookmarksAdapter
-    recycler_view.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
-
-    bookmarks_switch.isChecked = sharedPreferenceUtil.showBookmarksAllBooks
-    bookmarks_switch.setOnCheckedChangeListener { _, isChecked ->
-      bookmarkViewModel.actions.offer(Action.UserClickedShowAllToggle(isChecked))
-    }
-
-    compositeDisposable.add(bookmarkViewModel.effects.subscribe { it.invokeWith(this) })
-  }
+  override val showAllPagesSwitch: Switch by lazy { bookmarks_switch }
+  override val noItems: TextView by lazy { no_bookmarks }
+  override val recyclerView: RecyclerView by lazy { recycler_view }
+  override val layoutId: Int = R.layout.activity_bookmarks
+  override val title: String by lazy { getString(R.string.bookmarks) }
+  override val switchIsChecked: Boolean by lazy { sharedPreferenceUtil.showBookmarksAllBooks }
 
   override fun onCreateOptionsMenu(menu: Menu): Boolean {
     menuInflater.inflate(R.menu.menu_bookmarks, menu)
     val search = menu.findItem(R.id.menu_bookmarks_search).actionView as SearchView
     search.queryHint = getString(R.string.search_bookmarks)
```

Comment:
the onCreateOptionsMenu is the same aside from the query hint, that is all that should be an abstract val